_studio/mfx_lib/scheduler/include/mfx_scheduler_core_thread.h

```diff
@@ -37,16 +38,25 @@ struct MFX_SCHEDULER_THREAD_CONTEXT
 
     MFX_SCHEDULER_THREAD_CONTEXT() = delete;
 
-    MFX_SCHEDULER_THREAD_CONTEXT(mfxU32 id, SchedulerThreadFunc func)
+    MFX_SCHEDULER_THREAD_CONTEXT(mfxU32 id)
       : state(State::Waiting)
       , threadNum(id)
-      , threadHandle([this, func](){ func(this); })
+      , threadHandle([this, start_future=start.get_future()]() {
+          start_future.wait();
```

Comment:
if _SetScheduling_ fails we will have dangling thread here (ctx dtor will wait infinite at _threadHandle.join()_)

Reply:
I will check, thank you.

Reply:
done

Reply:
@dvrogozh, Using of identifier initializer at lambda's capture list:
`start_future=start.get_future()`
requires C++14. AFAIR we're restricted to C++11, aren't we?

Reply:
Yes, that's why I changed this place. See in the updated code.